### PR TITLE
Fix Microsoft.ReactNative.Cxx nuget package missing files

### LINKED ISSUE #9966: App build using NuGets fails with `Cannot open source file: '...\.nuget\packages\microsoft.reactnative.cxx\0.0.0-canary.499\bridging\LongLivedObject.cpp'`
### Problem Description

Trying to build against the nugets in `preview` or `canary` (0.69 and main) fails.

It looks like some new files have been added to the `Microsoft.ReactNative.CXX` project, but either the paths in the vcxitems are incorrect, or the files are missing when the `Microsoft.ReactNative.CXX` nuget is published.

### Steps To Reproduce

In a pwsh prompt:

```powershell
git clone https://github.com/microsoft/react-native-windows-samples.git
cd react-native windows-samples\samples\CalculatorNuGet
yarn
..\..\.github\scripts\UpgradeSmokeTest.ps1 canary -Force $True
npx react-native-windows-init --overwrite --experimentalNuGetDependency true --version canary
npx react-native run-windows --logging --no-packager --no-launch --arch x86                   
```

### Expected Results

_No response_

### CLI version

8.0.0-alpha.6

### Environment

```markdown
info Fetching system and libraries information...
System:
    OS: Windows 10 10.0.25115
    CPU: (8) x64 Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
    Memory: 4.46 GB / 15.84 GB
  Binaries:
    Node: 16.13.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.18 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 8.1.2 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Not Found
    Visual Studio: 16.11.32126.315 (Visual Studio Enterprise 2019)
  Languages:
    Java: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0
    react-native: 0.0.0-20220501-2027-d992ae044 => 0.0.0-20220501-2027-d992ae044
    react-native-windows: 0.0.0-canary.499 => 0.0.0-canary.499
  npmGlobalPackages:
    *react-native*: Not Found
info React Native v0.60.0 is now available (your project is running on v0.0.0-20220501-2027-d992ae044).
info Changelog: https://github.com/facebook/react-native/releases/tag/v0.60.0.
info Diff: https://react-native-community.github.io/upgrade-helper/?from=0.0.0-20220501-2027-d992ae044.
info To upgrade, run "react-native upgrade".
```

### Target Platform Version

10.0.19041

### Target Device(s)

Desktop

### Visual Studio Version

Visual Studio 2019

### Build Configuration

Debug

### Snack, code example, screenshot, or link to a repository

This was reproed in the CalculatorNuGet upgrade CI: https://github.com/microsoft/react-native-windows-samples/runs/6372520322?check_suite_focus=true

### PR DESCRIPTION
**Pull request by jonthysell**
The `Microsoft.ReactNative.Cxx.vcxitems` project is used in two places - from within the file structure of the RNW package (in this repo, or via npm) and also within the `Microsoft.ReactNative.Cxx` NuGet package, which has a different (flattened) file structure.

In addition, certain files were not included in the NuGet package, which would break C++ apps building against the binaries.

This PR updates the `Layout-MSRN-Headers.ps1` script to copy the missing files, and updates the `Microsoft.ReactNative.Cxx.vcxitems` project to use the correct paths in both use cases.

This PR also fixes a bug in the pipelines that caused the CLI tests creating new apps against the NuGets from being run, which would have caught this bug.

Fixing the pipeline bug also has the side effect of actually running the CLI tests running against the WinAppSDK template, which revealed other bugs which this PR also resolves.

Closes #9966

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9967)

### CODE REVIEW

File: vnext/Scripts/Tfs/Layout-MSRN-Headers.ps1 [hunk 1]
```diff
@@ -38,9 +38,9 @@ Copy-Item -Force -Path $ReactNativeRoot\ReactCommon\jsi\jsi\threadsafe.h -Destin
 # Microsoft.ReactNative.CXX project TurboModule files
 New-Item -ItemType Directory -Path $TargetRoot\Microsoft.ReactNative.Cxx\ReactCommon
 Copy-Item -Force -Path $ReactNativeRoot\ReactCommon\callinvoker\ReactCommon\CallInvoker.h -Destination $TargetRoot\Microsoft.ReactNative.Cxx\ReactCommon\
-Copy-Item -Force -Path $ReactNativeRoot\ReactCommon\react\bridging\LongLivedObject.cpp -Destination $TargetRoot\Microsoft.ReactNative.Cxx\ReactCommon\
 Copy-Item -Force -Path $ReactNativeRoot\ReactCommon\react\bridging\CallbackWrapper.h -Destination $TargetRoot\Microsoft.ReactNative.Cxx\ReactCommon\
-Copy-Item -Force -Path $ReactNativeRoot\ReactCommon\react\nativemodule\core\ReactCommon\LongLivedObject.h -Destination $TargetRoot\Microsoft.ReactNative.Cxx\ReactCommon\
+Copy-Item -Force -Path $ReactNativeRoot\ReactCommon\react\bridging\LongLivedObject.cpp -Destination $TargetRoot\Microsoft.ReactNative.Cxx\ReactCommon\
```
Comment (JunielKatarn) [inline vnext/Scripts/Tfs/Layout-MSRN-Headers.ps1:42]: (nit; unrelated)

This code is unnecessarily repetitive.
We can make this way more maintainable by using a loop.

I'll take a look at it on Friday.

**Review by JunielKatarn — APPROVED** [review]

File: vnext/Scripts/Tfs/Layout-MSRN-Headers.ps1 (hunk 1 above)
Comment (jonthysell) [reply vnext/Scripts/Tfs/Layout-MSRN-Headers.ps1:42]: Sure, but we don't copy every file, and in this case, we need to make sure to grab the "correct" `LongLivedObject.h`.

**Review by vmoroz — APPROVED** [review]
:shipit:

**Comment by jonthysell** [comment]
FYI @AgneLukoseviciute, it looks like the CLI tests for new WinAppSDK projects weren't running either, so this should fix those too.

**Pull request merged**